explainability/core/counterfactuals/src/main/java/org/kie/kogito/explainability/local/counterfactual/CounterFactualScoreCalculator.java

```diff
@@ -97,18 +78,37 @@ public BendableBigDecimalScore calculateScore(CounterfactualSolution solution) {
 
         try {
             List<PredictionOutput> predictions = predictionAsync.get();
-            final double distance = predictionDistance(predictions, solution.getGoal());
-            primaryHardScore -= distance;
-            logger.info("Penalise outcome (output: {})", distance);
+
+            final List<Output> outputs = predictions.get(0).getOutputs();
```

Comment:
not sure whether it makes sense to iterate and/or process / compose predictions asynchronously similarly to what it's done in `LimeExplainer`?

Reply:
@tteofili thanks for the comments! I think it makes perfect sense, as this implementation assumes:

1. Each prediction batch will have a single element (since we need to instantiate a counterfactual explainer for each one)
2. The score calculating method is implemented as sync, so a rethink is needed to better integrate with the async prediction request.

I agree that this should be improved, but I think it might be better to have a separate PR. wdyt?